test/retained_topic_map.cpp

```diff
@@ -206,39 +209,49 @@ BOOST_AUTO_TEST_CASE(erase_upper_first) {
     }
 }
 
-#if 0
-
 BOOST_AUTO_TEST_CASE(large_number_of_topics) {
-    retained_topic_map<std::size_t> map;
+    retained_topic_map< std::pair<std::size_t, std::size_t> > map;
```

Comment:
```suggestion
    retained_topic_map<std::pair<std::size_t, std::size_t>> map;
```

mqtt_cpp requires at least C++14, so we don't need to care about C++03 style.